.ci/scripts/build-environment/downstream-builder/generate_downstream.sh

```diff
@@ -101,71 +99,94 @@ elif [ "$COMMAND" == "downstream" ]; then
-    fi
+        bundle exec compiler.rb -a -e terraform -f validator -o $LOCAL_PATH/tfplan2cai -v $VERSION
 
-    if [ "$COMMAND" == "downstream" ]; then
-      pushd $LOCAL_PATH
-      go get -d github.com/hashicorp/terraform-provider-google-beta@$BASE_BRANCH
-      go mod tidy
-      set +e
-      make build
-      set -e
-      popd
-    fi
-elif [ "$REPO" == "tf-oics" ]; then
-    # use terraform generator with oics override
-    bundle exec compiler.rb -a -e terraform -f oics -o $LOCAL_PATH -v $VERSION
-else
-    if [ "$REPO" == "terraform" ]; then
-        if [ "$VERSION" == "ga" ]; then
-            bundle exec compiler.rb -a -e $REPO -o $LOCAL_PATH -v $VERSION --no-docs
-            bundle exec compiler.rb -a -e $REPO -o $LOCAL_PATH -v beta --no-code
-        else
-            bundle exec compiler.rb -a -e $REPO -o $LOCAL_PATH -v $VERSION
+        # Ignore 'tgc-base' step, because this is a new provider.
+        # TODO: remove this condition on next cai2hcl-related PR.
+        if [ "$COMMAND" != "base" ]; then
+        # Generate cai2hcl
+            pushd $LOCAL_PATH
+            rm -rf ./cai2hcl/*
+            popd
+
+            bundle exec compiler.rb -a -e terraform -f tgc_cai2hcl -o $LOCAL_PATH/cai2hcl -v $VERSION
         fi
-        pushd ../
-        make tpgtools OUTPUT_PATH=$LOCAL_PATH VERSION=$VERSION
 
-        # Only generate TeamCity-related file for TPG and TPGB
-        if [ "$VERSION" == "ga" ] || [ "$VERSION" == "beta" ]; then
-            make teamcity-servicemap-generate OUTPUT_PATH=$LOCAL_PATH VERSION=$VERSION
+        if [ "$COMMAND" == "downstream" ]; then
+          pushd $LOCAL_PATH
+          go get -d github.com/hashicorp/terraform-provider-google-beta@$BASE_BRANCH
+          go mod tidy
+          set +e
+          make build
+          set -e
+          popd
+        fi
+    elif [ "$REPO" == "tf-oics" ]; then
+        # use terraform generator with oics override
+        bundle exec compiler.rb -a -e terraform -f oics -o $LOCAL_PATH -v $VERSION
+    else
+        if [ "$REPO" == "terraform" ]; then
+            if [ "$VERSION" == "ga" ]; then
+                bundle exec compiler.rb -a -e $REPO -o $LOCAL_PATH -v $VERSION --no-docs
+                bundle exec compiler.rb -a -e $REPO -o $LOCAL_PATH -v beta --no-code
+            else
+                bundle exec compiler.rb -a -e $REPO -o $LOCAL_PATH -v $VERSION
+            fi
+            pushd ../
+            make tpgtools OUTPUT_PATH=$LOCAL_PATH VERSION=$VERSION
+
+            # Only generate TeamCity-related file for TPG and TPGB
+            if [ "$VERSION" == "ga" ] || [ "$VERSION" == "beta" ]; then
+                make teamcity-servicemap-generate OUTPUT_PATH=$LOCAL_PATH VERSION=$VERSION
+            fi
+            popd
         fi
-        popd
+    fi
+    popd
+else
+    if [ "$REPO" == "terraform-google-conversion" ]; then
+        make clean-tgc OUTPUT_PATH="$LOCAL_PATH"
+        make tgc OUTPUT_PATH="$LOCAL_PATH"
```

Comment:
I don't know tgc the best, does it mean that the following block is no longer needed?
```
    # Ignore 'tgc-base' step, because this is a new provider.
    # TODO: remove this condition on next cai2hcl-related PR.
    if [ "$COMMAND" != "base" ]; then
    # Generate cai2hcl
        pushd $LOCAL_PATH
        rm -rf ./cai2hcl/*
        popd

        bundle exec compiler.rb -a -e terraform -f tgc_cai2hcl -o $LOCAL_PATH/cai2hcl -v $VERSION
    fi
```

Reply:
I believe this was added for a similar reason to us adding the `if [ "$COMMAND" != "base" ]` .. to support backwards compatibility with the existing build.

Reply:
yes, the `if [ "$COMMAND" != "base" ]` conditional was added for backwards-compatibility with versions that didn't support cai2hcl generation yet.

Technically this should always run for `base` now - it's been around for a while.

Going forward, `make tgc` and `make clean-tgc` should probably handle cai2hcl as well.

Reply:
Okay, i refactored it to include these in the makefile. Could you take another look?